lighthouse-core/test/gather/computed/critical-request-chains-test.js

```diff
@@ -307,7 +312,35 @@ describe('CriticalRequestChain gatherer: extractChain function', () => {
       lastPathComponent: 'android-chrome-192x192.png',
     };
 
-    const criticalChains = CriticalRequestChains.extractChain(networkRecords);
+    const criticalChains = CriticalRequestChains.extractChain([networkRecords, mainResource]);
+    assert.deepEqual(criticalChains, {
+      0: {
+        request: constructEmptyRequest(),
+        children: {},
+      },
+    });
+  });
+
+  it('discards iframes as non-critical', () => {
+    const networkRecords = mockTracingData([HIGH, HIGH, HIGH], [[0, 1], [0, 2]]);
+    const mainResource = networkRecords[0];
+
+    // 1th record is the root document
+    networkRecords[0].url = 'https://example.com';
+    networkRecords[0].mimeType = 'text/html';
+    networkRecords[0]._resourceType._category = WebInspector.resourceTypes.Document._category;
+    // 2nd record is an iframe in the page
+    networkRecords[1].url = 'https://example.com/iframe.html';
+    networkRecords[1].mimeType = 'text/html';
+    networkRecords[1]._resourceType._category = WebInspector.resourceTypes.Document._category;
+    networkRecords[0].frameId = '2';
```

Comment:
is this intentional to keep changing frameId of the root document?

also is there a document request case that *is* within the same frame we should add to tests?

Reply:
Only html documents that are being fetched are Documents from iframe. XHR is not counted here